crates/ruff_index/src/list.rs

```diff
@@ -49,45 +121,43 @@ pub struct ListStorage<I, K, V = ()> {
 /// **Terminology**: The elements of a cons cell are usually called `head` and `tail` (assuming
 /// you're not in Lisp-land, where they're called `car` and `cdr`).  The elements of a snoc cell
 /// are usually called `rest` and `last`.
-///
-/// We use a tuple struct instead of named fields because we always unpack a cell into local
-/// variables:
-///
-/// ```ignore
-/// let ListCell(rest, last_key, last_value) = /* ... */;
-/// ```
 #[derive(Debug, Eq, PartialEq)]
-struct ListCell<I, K, V>(Option<I>, K, V);
+struct ListCell<K, V> {
+    rest: Option<ListCellId>,
+    key: K,
+    value: V,
+}
 
-impl<I: Idx, K, V> ListStorage<I, K, V> {
+impl<K, V> ListStorage<K, V> {
     /// Iterates through the entries in a list _in reverse order by key_.
-    pub fn iter_reverse(&self, list: Option<I>) -> ListReverseIterator<'_, I, K, V> {
+    #[allow(clippy::needless_pass_by_value)]
```

Comment:
`List` is `Copy`, so I'm not sure why clippy want us to always pass this around by value like we were before.  I guess the extra phantom fields increase the perceived complexity?